core/src/main/java/io/jenkins/pluginhealth/scoring/probes/JSR305Probe.java

```diff
@@ -0,0 +1,43 @@
+package io.jenkins.pluginhealth.scoring.probes;
+
+import java.util.List;
+
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+/**
+ * This probe checks for deprecated @Nonnull and @CheckForNull annotations in a plugin.
+ */
+@Component
+@Order(value = AbstractDeprecatedAnnotationImportsCheckProbe.ORDER)
+public class JSR305Probe extends AbstractDeprecatedAnnotationImportsCheckProbe {
+    public static final String KEY = "JSR-305";
+
+    @Override
+    public String key() {
+        return KEY;
+    }
+
+    @Override
+    public String getDescription() {
+        return "The probe checks for deprecated annotations.";
+    }
+
+    @Override
+    String getSuccessMessage() {
+        return "JSR305 annotation is updated";
```

Comment:
Or it is just not found right?

Reply:
I did not use the word "not found" here because it gives a negative connotation.

Like, the deprecated annotations should be found but they are not. 

But this probe is successful when they are not found.

Reply:
What I mean is perhaps JSR305 was never used in the plugin in the first place, so reading a message saying it was updated could be confusing.